src/Tools/AnalyzerRunner/IncrementalAnalyzerRunner.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FindSymbols.SymbolTree;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.IncrementalCaches;
+using Microsoft.CodeAnalysis.Shared.Options;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.CodeAnalysis.SolutionSize;
+using Microsoft.CodeAnalysis.Storage;
+
+namespace AnalyzerRunner
+{
+    internal class IncrementalAnalyzerRunner
+    {
+        private readonly Options _options;
+
+        public IncrementalAnalyzerRunner(Options options)
+        {
+            _options = options;
+        }
+
+        public bool HasAnalyzers => _options.IncrementalAnalyzerNames.Any();
+
+        internal async Task RunAsync(Workspace workspace, CancellationToken cancellationToken)
+        {
+            if (!HasAnalyzers)
+            {
+                return;
+            }
+
+            workspace.Options = workspace.Options
+                .WithChangedOption(StorageOptions.SolutionSizeThreshold, _options.UsePersistentStorage ? 0 : int.MaxValue)
+                .WithChangedOption(RuntimeOptions.FullSolutionAnalysis, _options.FullSolutionAnalysis)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.CSharp, _options.FullSolutionAnalysis)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.VisualBasic, _options.FullSolutionAnalysis);
+
+            if (!string.IsNullOrEmpty(_options.ProfileRoot))
+            {
+                ProfileOptimization.StartProfile(nameof(IIncrementalAnalyzer));
```

Comment:
What does `ProfileOptimization.StartProfile` do?

Reply:
Enables multi-core JIT to help reduce the impact of JIT on repeated measurements.